cmd/nvidia-dra-controller/imex.go

```diff
@@ -105,19 +107,29 @@ func (m *ImexManager) manageResourceSlices(ctx context.Context, owner resourcesl
 		return fmt.Errorf("error starting resource slice controller: %w", err)
 	}
 
+	imexChannelPool := make(map[string]int)
```

Comment:
Since we can't rely solely on the `cliqueid` to calculate an offset for the start of the channels, we need an extra level of indiraction here. 

As I mentioned in the call the other day, I think we can make it:
```
imexChannelOffsets := make(map[string]map[string]int)
```
With the first element being the imex-id, the second being the clique-id, and the third being the offset for where to start the channel count for that imex-id, clique-id combination.

The trick will be to look through that map each time we need to assign a new offset to choose the lowest one available for the given imex-id, clique-id combination.

We also need to set a max limit of 2048 (thats a limit in the driver) so that we never create more than that many channels in total. If there are so many cliques that we would be forced to go beyond that, then we need to throw an error.